src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPOperations.java

```diff
@@ -174,4 +185,52 @@ public void delete(@Connection MailboxConnection connection,
                                       e);
     }
   }
+
+  /**
+   * Moves the email with id {@code emailId} from the sourceFolder to the {@code emailId} to the {@code emailId}.
+   * <p>
+   *
+   * @param connection          The corresponding {@link MailboxConnection} instance.
+   * @param sourceFolder        Mailbox folder where the emails to be moved are located.
+   * @param targetFolder        Mailbox folder where the emails will be moved.
+   * @param emailId             Email ID Number of the email to move.
+   * @param createTargetFolder  Create the target folder if it does not exist.
+   */
+  @Summary("Moves an email from the given source mailbox folder to the target mailbox folder")
+  @Throws(EmailMarkingErrorTypeProvider.class)
+  public void moveToFolder(@Connection MailboxConnection connection,
+                           @Optional(defaultValue = INBOX_FOLDER) @OfValues(MailboxFolderValueProvider.class) String sourceFolder,
+                           @OfValues(MailboxFolderValueProvider.class) String targetFolder,
+                           @Summary("Email ID Number of the email to move") @DisplayName("Email ID") long emailId,
+                           @Summary("Create the target folder if it does not exist") @Optional(
+                               defaultValue = "false") boolean createTargetFolder) {
+    try {
+      Folder defaultFolder = connection.getDefaultFolder();
+      IMAPFolder destinationFolder = (IMAPFolder) defaultFolder.getFolder(targetFolder);
+      if (!destinationFolder.exists()) {
+        if (createTargetFolder) {
+          destinationFolder.create(Folder.HOLDS_FOLDERS | Folder.HOLDS_MESSAGES);
+        } else {
+          throw new FolderNotFoundException(destinationFolder);
+        }
+      }
+
+      IMAPFolder sourceMailboxFolder = (IMAPFolder) connection.getFolder(sourceFolder, READ_WRITE);
+      javax.mail.Message message = sourceMailboxFolder.getMessageByUID(emailId);
```

Comment:
Could the sourceMailboxFolder be null? if so, a null check should be in place and throw an FolderNotFoundException on sourceFolder.